chapters/chapter2/3-random_forest.qmd

```diff
@@ -127,29 +143,73 @@ Il s'agit d'un cas particulier où le nombre de sous-échantillons est égal à
 
 ### Choix des hyper-paramètres du modèle
 
-L'estimation Out-of-Bag (OOB) et la validation croisée sont deux méthodes clés pour optimiser les hyper-paramètres d'une forêt aléatoire. Les deux approches permettent de comparer différentes combinaisons d'hyper-paramètres et de sélectionner celles qui maximisent les performances prédictives, l'OOB étant souvent plus rapide et moins coûteuse, tandis que la validation croisée est plus fiable dans des situations où le surapprentissage est un risque important.
+L'estimation Out-of-Bag (OOB) et la validation croisée sont deux méthodes clés pour optimiser les hyper-paramètres d'une forêt aléatoire. Les deux approches permettent de comparer les performances obtenues pour différentes combinaisons d'hyper-paramètres et de sélectionner celles qui maximisent les performances prédictives, l'OOB étant souvent plus rapide et moins coûteuse, tandis que la validation croisée est plus fiable dans des situations où le surapprentissage est un risque important.
+
+Il convient néanmoins de mettre en place une stratégie d'optimisation des hyperparamètres. Plusieurs stratégies existent pour y parvenir, les principales sont exposées dans ce qui suit.
+
+Il convient de noter que les implémentations "clés en main" en R ou Python ne permettent en général d'optimiser que les hyper-paramètres qui influent le plus sur la performance du modèle. Il s'agit généralement de mtry (nombre de variables sélectionnées aléatoirement à chaque nœud), la taille minimale des nœuds (nodesize ou min_node_size), et la taille de l'échantillon utilisé pour construire chaque arbre (sample size).
+
+
+
+#### Méthodes de recherche exhaustives
+
+- **Recherche sur grille** (Grid Search): Cette approche simple explore toutes les combinaisons possibles d'hyperparamètres définis sur une grille. Les paramètres continus doivent être discrétisés au préalable. La méthode est exhaustive mais coûteuse en calcul, surtout pour un grand nombre d'hyperparamètres.
+
+- **Recherche aléatoire** (Random Search): Plus efficace que la recherche sur grille, cette méthode échantillonne aléatoirement les valeurs des hyperparamètres dans un espace défini. Bergstra et Bengio (2012) ont démontré sa supériorité pour les réseaux neuronaux, et elle est également pertinente pour les forêts aléatoires. La distribution d'échantillonnage est souvent uniforme.
+
+
+#### Optimisation séquentielle/itérative basée sur un modèle (SMBO)
+
+La méthode SMBO (Sequential model-based optimization) est une approche plus efficace que les précédentes car elle s'appuie sur les résultats des évaluations déjà effectuées pour guider la recherche des prochains hyper-paramètres à tester. 
+
+Voici les étapes clés de cette méthode:
```

Comment:
Tu as le papier de référence sur le SMBO?

```suggestion
La méthode SMBO (_Sequential model-based optimization_) est une approche plus efficace que les précédentes car elle s'appuie sur les résultats des évaluations déjà effectuées pour guider la recherche des prochains hyper-paramètres à tester. Voici les principales étapes de cette méthode:
```

Reply:
Probst et al 2019: "Sequential model-based optimization

Sequential model-based optimization (SMBO) is a very successful tuning strategy that iteratively tries to find the best
hyperparameter settings based on evaluations of hyperparameters that were done beforehand. SMBO is grounded in the “black-box function optimization” literature (Jones et al., 1998) and achieves state-of-the-art performance for solving a number of optimization problems (Hutter et al., 2011)."